src/OneTimeTokenProvider.js

```diff
@@ -0,0 +1,39 @@
+import Connection from './Connection'
+
+export default class OneTimeTokenProvider {
+  constructor(publicAPIKey, options) {
+    this.publicAPIKey = publicAPIKey
+    this.options = options
+    this.token = null
+    this.timeout = null
+    this.marginToExpire = 0.05 // a margin to prevent unexpected expiration (5% of the time)
+  }
+
+  get() {
+    return this.retrieve()
+      .then(() => {
+        const token = this.token
+        this.token = null
+        return token
+      })
+  }
+
+  retrieve() {
+    if (this.token != null) {
+      return Promise.resolve()
+    }
+    return this.issue()
+      .then(data => this.storeToken(data))
+  }
+
+  issue() {
+    return Connection.build(true, Object.assign({}, this.options, { apiKey: this.publicAPIKey }))
+      .sendJson('/auth/tokens', { type: 'one-time' })
+      .then(response => response.data)
+  }
+
+  storeToken(data) {
+    this.timeout = setTimeout(() => { this.token = null }, data.expires_in * (1 - this.marginToExpire) * 1000)
```

Comment:
I think a call to `clearTimeout(this.timeout)` is missing here.

Reply:
Ah, I totally forgot about it! Though I stored the return value from `setTimeout`...
I added the change. 👍